Networking Demo AppTests/APIClientTests.swift

```diff
@@ -77,5 +77,41 @@ class APIClientTests: XCTestCase {
         }
         waitForExpectations(timeout: 10, handler: nil)
     }
+    
+    @available(iOS 15.0.0, *)
+    func testAsyncValidCreateRequest() async throws {
+        let apiClientExpectation = expectation(description: "expect apiClient call back to be called.")
```

Comment:
why do we even have expectation for this? :-D

Reply:
technically we don't, but i wanted to assert at least something to check that the test works as intended in line 94. otherwise it would be just `break`.

Reply:
maybe assert the response is of the expected type, or just a break, why not? since the `.failure` case has `XCTFail()`. but I prefer to assert something about the response object.